### ISSUE TITLE
Meldungen der Browser-Konsole im ioBroker.log anzeigen

### ISSUE THREAD

**Issue by Zefau** [Zefau/ioBroker.jarvis#2550]

**Comment by Zefau** [comment]
@mcuiobroker ich hab hierfür mal ein Feature Request angelegt, aber hattest du das nicht schon irgendwo erwähnt?

**Comment by Zefau** [comment]
ab alpha-46 neue Option:

![grafik](https://github.com/Zefau/ioBroker.jarvis/assets/14368811/4acc44aa-4032-4fd8-b69c-7586d11e9559)

**Comment by Zefau** [comment]
Dieser Feature Request wurde mit `v3.2.0-alpha.46` implementiert. Bitte bestätigen und auf Fehler prüfen. Wenn alles in Ordnung ist, gerne das Issue schließen.<br /><hr />This feature request has been implemented with `v3.2.0-alpha.46`. Please verify and test the feature for any bugs. If everything works as expected, you may close the issue.

**Comment by mcuiobroker** [comment]
Sehr gut.
Es kommen anscheinend nur die warn-Meldungen. Es sollten alle Meldungen kommen auch die info und error-Meldungen. 
Die Anfangsmeldungen von jarvis slebst Ladezeiten bekommt man nicht, da socket nocht nicht aufgebaut ist?
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/26949441-dc38-420a-8687-611d5dad2378)
Diese meldungen kommen häufig. Auch wenn man das Frontend offen hat.

**Comment by Zefau** [comment]
>  Es kommen anscheinend nur die warn-Meldungen. Es sollten alle Meldungen kommen auch die info und error-Meldungen.

Bei mir in der Dev-Umgebung kommen alle, also ist nicht auf Warnungen beschränkt. Aber ja, erst wenn der Socket steht werden diese übertragen, also die ersten Meldungen schaffen es quasi nicht ins ioBroker.

>  Diese meldungen kommen häufig. Auch wenn man das Frontend offen hat.

Kommt davor ein disconnect wegen timeout?

**Comment by mcuiobroker** [comment]
Jetzt kommen auch die normalen info.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/88cacd7d-01b7-40a9-ac83-384f29d8e6e1)

![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/d39dc8e0-bdf2-49b3-bc1e-30daeeb19755)

ok. Ich beobachte mal. Mache aber schon mal zu. Danke.

**Comment by mcuiobroker** [comment]
Also die console.log Ausgabe von scripts werden nicht angezeigt

**Comment by Zefau** [comment]
Geht bei mir mit 
```js
function test() {
   console.warn("TEEEEEST");
}

test();
```

![grafik](https://github.com/Zefau/ioBroker.jarvis/assets/14368811/53ddc960-b853-43ec-a696-8b5593757b57)

**Comment by mcuiobroker** [comment]
Auch auf dem Handy?

**Comment by Zefau** [comment]
Das hab ich jetzt nicht getestet, aber wer weiß, ob und wie `console` da überhaupt implementiert ist

**Comment by mcuiobroker** [comment]
console.warn kommt aber console.log nicht

**Comment by Zefau** [comment]
Sonst kannst du es auch "manuell" via socket aufrufen: `log(criticality, messages)`

**Comment by mcuiobroker** [comment]
log('info', 'TESTTETSTETETST')?
funktioniert nicht

**Comment by Zefau** [comment]
Über den Socket, also `Socket.log`

**Comment by Zefau** [comment]
ab alpha-48 ist es `Socket.log(messages, criticality)`, wenn `criticality` nicht angegeben wird (also nur Socket.log(messages)`), ist es `info` per default.

**Comment by mcuiobroker** [comment]
`window.Socket.log('info', 'TESTTETSTETETST')`
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/e47a5e8a-5174-4965-b4ea-b9c3842aec93)
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/112c7b45-b735-4c05-a4d0-4fdee3d40d55)

**Comment by Zefau** [comment]
ah, `console.log` fehlt, stimmt.

**Comment by Zefau** [comment]
mit `v3.2.0-alpha.49` dann auch inkl. `console.log` (hatte fälschlicherweise `console.info` drin).

**Comment by Zefau** [comment]
Dieser Feature Request wurde mit `v3.2.0-alpha.48` implementiert. Bitte bestätigen und auf Fehler prüfen. Wenn alles in Ordnung ist, gerne das Issue schließen.<br /><hr />This feature request has been implemented with `v3.2.0-alpha.48`. Please verify and test the feature for any bugs. If everything works as expected, you may close the issue.

**Comment by Zefau** [comment]
Dieser Feature Request wurde mit `v3.2.0-alpha.49` implementiert. Bitte bestätigen und auf Fehler prüfen. Wenn alles in Ordnung ist, gerne das Issue schließen.<br /><hr />This feature request has been implemented with `v3.2.0-alpha.49`. Please verify and test the feature for any bugs. If everything works as expected, you may close the issue.

**Comment by mcuiobroker** [comment]
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/25ee302b-56ea-4590-bc3a-3b79a2d4586d)

**Comment by mcuiobroker** [comment]
Jetzt sieht es so aus, als wenn das log von jarvis bekommt, auch wenn die Option für den client nicht angeschaltet ist.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/70f82aa0-3776-4ba0-b92b-ac659dfb0525)
Vorher habe ich immer gesehen von welchem Script das Log kam.

**Comment by Zefau** [comment]
Verstehe nicht, was du meinst. Du bist in der Browser-Konsole? Du meinst, der Eintrag kommt nicht von dir? Oder was genau?

**Comment by mcuiobroker** [comment]
Der Eintrag kommt von mir. Normalerweise steht dann dort nicht `index-24....js:134` sondern z.B.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/d2124d12-9bac-4b89-a5cc-1f71913db062)
Je nachdem welches Script den Log verursacht hat.

**Comment by Zefau** [comment]
Jo, das liegt jetzt daran, dass ich `console` überschreibe mit einer eigenen Funktion, die dann das Log in das Backend schickt und selbst auch wieder in die Konsole schreibt. Daher ist der Pfad verändert.

Wüsste nicht, wie man das umgeht, wenn `console` überschrieben wird.

**Comment by mcuiobroker** [comment]
Kann man das nicht nur für die clients einstellen, bei denen die Option eingestellt wird?

**Comment by Bernd9000** [comment]
Kann ich das `future` aktuell für alle Clients Global auf true einstellen ?
Vielleicht später mal als Einstellmöglichkeit in der jarvis Konfiguration ?

**Comment by Bernd9000** [comment]
:-)
Mit `future` meinte ich natürlich das hier angelegte `feature`
War nur so ne Idee, mal eben zum Test den Schalter umlegen in der jarvis Konfiguration und dann im IOB Log die Fehler ablesen und kopieren. Jetzt muß ich ja erstmal in der IOB Datenbank schauen mit welchem Client bin ich eigentlich gerade aktiv und dann auf true stellen.Dann teste ich mit einem anderen Browser/Client und muß wieder dort schauen und den DP auf true stellen.
Wäre vielleicht für Anfänger einfacher ?

**Comment by Zefau** [comment]
Mit der nächsten alpha könnt ihr das Log-Level auswählen, was im ioBroker angezeigt werden soll. Ihr müsst dazu allerdings den bisherigen Datenpunkt `browserConsoleInLog` löschen, damit er korrekt neu angelegt wird. Außerdem gibt es einen neuen Datenpunkt `jarvis.0.clients.allBrowserConsoleInLog`, der global alle Browser-Ausgaben steuert (und die einzelnen Einstellungen übersteuert):

![grafik](https://github.com/Zefau/ioBroker.jarvis/assets/14368811/6761532a-bde3-4efd-afb1-12f14767bc43)

**Comment by Zefau** [comment]
Dieser Feature Request wurde mit `v3.2.0-alpha.56` implementiert. Bitte bestätigen und auf Fehler prüfen. Wenn alles in Ordnung ist, gerne das Issue schließen.<br /><hr />This feature request has been implemented with `v3.2.0-alpha.56`. Please verify and test the feature for any bugs. If everything works as expected, you may close the issue.

**Comment by mcuiobroker** [comment]
Passt. Danke.
@Bernd9000 ?

**Comment by Bernd9000** [comment]
Danke,
